site/public/js/drag-and-drop.js

```diff
@@ -1051,3 +1056,49 @@ function handleUploadCourseMaterials(csrf_token, expand_zip, cmPath, requested_p
         }
     });
 }
+
+/**
+ * @param csrf_token
+ */
+
+function handleEditCourseMaterials(hide_from_students, requested_path, sections) {
+    console.log("TESTInEdit");
+    var edit_url = buildCourseUrl(['course_materials', 'edit']);
+    console.log(edit_url);
+    var return_url = buildCourseUrl(['course_materials']);
+    var formData = new FormData();
+
+    formData.append('hide_from_students', hide_from_students);
+    formData.append('requested_path', requested_path);
+    if(sections !== null){
+        formData.append('sections', sections);
+    }
+
+    $.ajax({
+        url: edit_url,
+        data: formData,
```

Comment:
The `formData` here does not have csrf token. 

The router will check csrf token if the method is POST. If the check fails, in the case of web access, you will be redirected to the homepage. When I first designed this (https://github.com/Submitty/Submitty/pull/3980#discussion_r301152578), I assumed that it is a rare case that a normal user doesn`t have a csrf token unless they are trying to attack Submitty. However, this behavior may be confusing to developers who accidentally forget POSTing CSRF tokens.

I will open another issue to address this. For your case, appending csrf token to the formData will be fine.